thread-rcu/main.c

```diff
@@ -0,0 +1,74 @@
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "rcu.h"
+
+struct test {
+    unsigned int count;
+};
+
+static struct test __rcu *foo;
```

Comment:
Use `dut` instead of `foo` generally. Here `dut` means [device under test](https://en.wikipedia.org/wiki/Device_under_test).